packages/neuron-wallet/src/services/addresses.ts

```diff
@@ -445,4 +449,36 @@ export default class AddressService {
   public static async deleteByWalletId(walletId: string): Promise<void> {
     await getConnection().createQueryBuilder().delete().from(HdPublicKeyInfo).where({ walletId }).execute()
   }
+
+  public static async getPrivateKeyByAddress({
+    walletID,
+    password,
+    address,
+  }: {
+    walletID: string
+    password: string
+    address?: string
+  }): Promise<string> {
+    const wallet = WalletService.getInstance().get(walletID)
+
+    const addresses = await AddressService.getAddressesByWalletId(walletID)
+    let addr = address ? addresses.find(addr => addr.address === address) : addresses[0]
+
+    if (!addr) {
+      const usedBlake160s = new Set(await AssetAccountService.blake160sOfAssetAccounts())
+      addr = addresses.find(a => !usedBlake160s.has(a.blake160))!
```

Comment:
Does it have a condition that args in the `AssetAccount` entity but not in the `HdPublicKeyInfo` entity?